platform/engineering/design-docs/2020-04-03-separate-content-build.md

```diff
@@ -0,0 +1,486 @@
+      question](#react-application-landing-pages), this may mean publishing the
+      page in Drupal or switching the flag in the JSON file
+1. Deploy a new content build
+
+## Specifics
+
+### Detailed Design
+
+#### Content build in a separate repository
+The content build will live in a separate repository called `content-build`.
+This repository will contain only code pertaining to the content build.
+
+![Separated content code base](images/separated-content-build/phase-2-separated-content-build.png)
+
+#### Triggering builds
+**Applications**  
+The `vets-website` build is triggered in CircleCI when:
+1. Code is pushed to a feature branch on GitHub
+1. A feature branch is merged into `master`
+1. A weekdaily deploy is triggered
+    - This build results in a versioned release artifact
+
+**Content**  
+The content build is triggered in Jenkins when an editor with the required
+permissions initiates a content deploy.
+
+#### Coordinating deployments to S3
+We use S3 to serve a static site. To do this, the bucket consists of a file tree
+which corresponds to the URLs on VA.gov. For example, `/education/index.html` is
+the file that's served at www.va.gov/education/.
+
+The output of the application build **currently** live in the `/generated/`
+directory. To support the transition, the **new** application build will put its
+assets in the `/applications/` directory.
+
+![Coordinating deployments to S3](images/separated-content-build/coordinating-deployments-to-s3.png)
+
+To ensure that one build won't override files in the other, we'll have to
+1. Update
+   [`vets-website-deploy.sh`](https://github.com/department-of-veterans-affairs/devops/blob/2c63e6d62fa5ad2b6b87478e30d0a7bc2820c252/ansible/deployment/roles/deploy-vets-website/files/vets-website-deploy.sh)
+   to copy only to the `/applications/` directory
+1. Create a new deploy script for the content deployment which copies everything
+   over _except_ the `/applications/` directory
+
+#### Testing changes to applications
+- The Webpack configuration will use
+  [`html-webpack-plugin`](https://github.com/jantimon/html-webpack-plugin) to
+  generate temporary landing pages for React applications
+  - This will happen for all build types
+    - Engineers will be able to access the site on a `localhost` build
+    - CI can run E2E tests against production code without the output of the
+      content build
+- The E2E test script in CI will start a process to serve these pages so the
+  tests can run a browser and access the applications
+- **Important note:** These HTML files will not be served in production; the
+  landing pages for the applications come from the content build
+  - To ensure these pages aren't made in production, the deploy script will copy
+    only the files in `/applications/`
+
+#### Miscellaneous
+- We need a separate content validation job that runs once a day on a schedule
+  - **Question:** How will we validate links to `vets-website` assets?
```

Comment:
I think the answer is that we'll have a Docker container in Nomad which can `s3 sync` down all the files in production and run all content validation checks on that local copy.

Does this sound feasible, @wyattwalter ?

Will we need an interim solution until Nomad is ready to be used?